test/metabase/search/appdb/index_test.clj

```diff
@@ -425,39 +428,51 @@
-                                                                  :value_ref  (mt/$ids :products $title)
-                                                                  :schedule   "0 0 0 * * *"
-                                                                  :state      "initial"
-                                                                  :creator_id (mt/user->id :rasta)}]
-        ;; :model/ModelIndexValue does not have id column so does not work nicely with with-temp
-        (t2/insert! :model/ModelIndexValue {:name       entity-name
-                                            :model_index_id model-index-id
-                                            ;; :model_id = model-id
-                                            :model_pk   42})
-        (is (=? (index-entity
-                 {:model            "indexed-entity"
-                  :model_id         42
-                  :name             entity-name
-                  :database_id      (mt/id)
-                  :collection_id    coll-id})
-                (ingest-then-fetch! "indexed-entity" entity-name)))))))
+  (let [model-id (fn [miv] (str (:model_index_id miv) ":" (:model_pk miv)))]
+    (search.tu/with-temp-index-table
+      (mt/with-temp [:model/Collection      {coll-id :id} {}
+                     :model/Card            model         (assoc (mt/card-with-source-metadata-for-query
+                                                                  (mt/mbql-query products {:fields [$id $title]
+                                                                                           :limit  1}))
+                                                                 :type          "model"
+                                                                 :database_id   (mt/id)
+                                                                 :collection_id coll-id)
+                     :model/ModelIndex      model-index   {:model_id   (:id model)
+                                                           :pk_ref     (mt/$ids :products $id)
+                                                           :value_ref  (mt/$ids :products $title)
+                                                           :schedule   "0 0 0 * * *"
+                                                           :state      "initial"
+                                                           :creator_id (mt/user->id :rasta)}]
+        (model-index/add-values! model-index)
+        (let [miv (t2/select-one :model/ModelIndexValue :model_index_id (:id model-index))]
+          (testing "Adding values indexes them"
+            (is (=? (index-entity
+                     {:model         "indexed-entity"
+                      :model_id      (model-id miv)
+                      :name          (:name miv)
+                      :database_id   (mt/id)
+                      :collection_id coll-id})
+                    (fetch-one "indexed-entity" :name (:name miv)))))
```

Comment:
What do you think about just using `fetch` here, so the test asserts that there is only one entry?